// @fluentui/react-motion-components-preview/lib-commonjs/components/Fade/Fade.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Fade: function() {
        return Fade;
    },
    FadeRelaxed: function() {
        return FadeRelaxed;
    },
    FadeSnappy: function() {
        return FadeSnappy;
    },
    createFadePresence: function() {
        return createFadePresence;
    }
});
const _reactmotion = require("@fluentui/react-motion");
const createFadePresence = ({ enterDuration = _reactmotion.motionTokens.durationNormal, enterEasing = _reactmotion.motionTokens.curveEasyEase, exitDuration = enterDuration, exitEasing = enterEasing } = {})=>({
        enter: {
            duration: enterDuration,
            easing: enterEasing,
            keyframes: [
                {
                    opacity: 0
                },
                {
                    opacity: 1
                }
            ]
        },
        exit: {
            duration: exitDuration,
            easing: exitEasing,
            keyframes: [
                {
                    opacity: 1
                },
                {
                    opacity: 0
                }
            ]
        }
    });
const Fade = (0, _reactmotion.createPresenceComponent)(createFadePresence());
const FadeSnappy = (0, _reactmotion.createPresenceComponent)(createFadePresence({
    enterDuration: _reactmotion.motionTokens.durationFast
}));
const FadeRelaxed = (0, _reactmotion.createPresenceComponent)(createFadePresence({
    enterDuration: _reactmotion.motionTokens.durationGentle
}));
